src/compiler/checker.ts

```diff
@@ -428,6 +429,22 @@ namespace ts {
             // return undefined if we can't find a symbol.
         }
 
+        /**
+         * Get symbols that represent parameter-property-declaration as parameter and as property declaration
+         * @param parameter a parameterDeclaration node
+         * @param parameterName a name of the parameter to get the symbols for.
+         * @return a tuple of two symbols 
+         */
+        function getSymbolsOfParameterPropertyDeclaration(parameter: ParameterDeclaration, parameterName: string): [Symbol, Symbol] {

```

Comment:
I wonder if the return type here should make it explicit over which is which (i.e.`{ parameterSymbol: Symbol; propertySymbol: Symbol; }`). Perhaps that's overkill.